extensions/2.0/Vendor/KITTYCAD_boundary_representation/README.md

```diff
@@ -0,0 +1,250 @@
+
+### Shell
+
+A _shell_ is a collection of _faces_ in 3D space which form a 'watertight' volume. A shell is represented by the `Shell` data structure.
+
+### Face
+
+A _face_ is a portion of a _surface_ in 3D space. Multiple faces are referenced to form _shells_.
+
+Faces are associated with the 3D surface they are defined on. The boundary of faces are defined by a set of _edges_ which form a closed _loop_ in 3D space, following the contours of its associated surface. The boundary of a face is defined by its _outer loop_. In a similar manner to _shells_, faces may have internal hollow regions; such regions are called _inner loops_. Both outer loops and inner loops are represented by the `Loop` data structure.
+
+![face-example](figures/face.png)
+
+#### Example
+
+This face is formed by a closed loop of 3D curves which are coincident to its associated planar surface. The grid visualizes the associated planar surface.
+
+### Loop
+
+A _loop_ is a set of _edges_ that combine to form a closed circuit. Edges are referenced in the loop so that their direction may be reversed.
+
+![loop-example](figures/loop.png)
+
+### Curve
+
+A _curve_ defines a function that evaluates a single parameter to a point in either 2D or 3D space, depending on context.
+
+Curves are referenced by _edges_ to define the boundary between _faces_ in 3D space. Supplementary curves may be referenced by _loops_ to define the same boundary between _faces_ in 2D _surface_ parameter space.
+
+#### Base structure
+
+```json
+{
+  "type": "curveType",
+  "curveType": {
+    "curveSpecificValue": 1
+  },
+  "domain": {
+    "min": 0,
+    "max": 1
+  }
+}
+```
+
+#### Line (start & direction)
+
+```json
+"line": {
+  "start": [-1, -1, -1],
+  "direction": [1, 1, 1],
+}
+```
+
+#### Line (start & end)
+
+
+```json
+"line": {
+  "start": [-1, -1, -1],
+  "end": [1, 1, 1],
+}
+```
+
+#### Circle
+
+
+```json
+"circle": {
+  "radius": 1,
+  "origin": [0, 0, 0],
+  "normal": [0, 0, 1],
+  "xbasis": [1, 0, 0]
+}
+```
+
+#### [NURBS](https://en.wikipedia.org/wiki/Non-uniform_rational_B-spline)
+
+
+```json
+"nurbs": {
+  "controlPoints": [
+    [1.0, 0.0, 0.0, 1.0],
+    [1.0, 1.0, 0.0, 0.7071],
+    [0.0, 1.0, 0.0, 1.0]
+  ],
+  "knotVector": [0, 0, 0, 1, 1, 1],
+  "order": 3
+}
+```
+
+## Surface
+
+A _surface_ defines a function that evaluates two parameters to a point in 3D space.
+
+Surfaces are referenced by _faces_ to define the implicit geometry bounded by _loops_.
+
+#### Base structure
+
+```json
+{
+  "type": "surfaceType",
+  "surfaceType": {
+    "surfaceSpecificValue": 1
+  },
+  "domain": {
+    "min": [0, 0],
+    "max": [1, 1]
+  }
+}
+```
+
+#### Plane
+
+```json
+"plane": {
+  "normal": [0, 0, 1],
+  "point": [0, 0, 0]
+}
+```
+
+#### Cylinder (extruded circle)
+
+```json
+"cylinder": {
+  "circle":  {
+    "radius": 1,
+    "origin": [0, 0, 0],
+    "normal": [0, 0, 1],
+    "xbasis": [1, 0, 0]
+  }
+}
+```
+
+#### Torus (revolved circle)
```

Comment:
Surfaces of revolution and extrusion are planned and in progress at the moment but haven't been added to this PR yet.

This particular section was intended to introduce a type of a parametric surface. Perhaps that could be made clearer.

Now that you mention it, I do wonder if it's worth having distinct types for cylinders and tori at all since they could be simply inferred as circle + extrusion/revolution.